src/RegisterPage.cpp

```diff
@@ -93,6 +94,7 @@ RegisterPage::RegisterPage(QWidget *parent)
 
         server_input_ = new TextField();
         server_input_->setLabel(tr("Homeserver"));
+        server_input_->setRegexp(QRegularExpression("[a-z0-9.-]+"));
```

Comment:
I think this will actually lead to issues with idn domains, i.e. ones ending in emoji. But we don't handle those correctly right now anyway...

Reply:
Ok, shall I just check it's not empty? I think that's what was done before, and if the server isn't reachable errors will be shown later :-)

Reply:
Yeah, that sounds reasonable

Reply:
That all good now?

Reply:
Yep, looks good. I need to test it manually still, then I think I can merge it :3